src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -628,6 +628,49 @@ public bool Remove(TKey key)
             return false;
         }
 
+        public bool Remove(TKey key, out TValue value)
+        {
+            if (key == null)
+            {
+                value = default(TValue);
```

Comment:
> Although variables passed as out arguments do not have to be initialized before being passed, the called method is required to assign a value before the method returns.

This is only applicable when the method returns to its caller; if it instead throws an exception, this does not apply.